lib/src/rules/unnecessary_breaks.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r"Don't use explicit `breaks`s when a break is implied.";
+
+const _details = r'''
+Only use a `break` in a non-empty switch case statement if you need to break
+before the end of the case body.  Dart does not support fallthrough execution
+for non-empty cases, so `break`s at the end of non-empty switch case statements
+are unnecessary.
+
+**BAD:**
+```dart
+switch (1) {
+  case 1:
+    print("one");
+    break;
+  case 2:
+    print("two");
+    break;
+}
+```
+
+**GOOD:**
+```dart
+switch (1) {
+  case 1:
+    print("one");
+  case 2:
+    print("two");
+}
+```
+
+```dart
+switch (1) {
+  case 1:
+  case 2:
+    print("one or two");
+}
+```
+
+```dart
+switch (1) {
+  case 1:
+    break;
+  case 2:
+    print("just two");
+}
+```
+''';
+
+class UnnecessaryBreaks extends LintRule {
```

Comment:
Consider defining a lint code from the outset that has both a problem and correction message.